src/core/ModuleManager.sol

```diff
@@ -144,17 +148,20 @@ abstract contract ModuleManager is AccountBase, Receiver {
         virtual
         returns (address[] memory array, address next)
     {
-        SentinelListLib.SentinelList storage _executors = _getModuleManagerStorage()._executors;
-        return _executors.getEntriesPaginated(cursor, size);
+        SentinelListLib.SentinelList storage $executors = $moduleManager().$executors;
+        return $executors.getEntriesPaginated(cursor, size);
     }
 
     /////////////////////////////////////////////////////
     //  Manage Fallback
     ////////////////////////////////////////////////////
 
-    function _installFallbackHandler(address handler, bytes calldata initData) internal virtual {
-        if (_isFallbackHandlerInstalled()) revert();
-        _getModuleManagerStorage().fallbackHandler = handler;
+    function _installFallbackHandler(address handler, bytes calldata params) internal virtual {
+        (bytes4 functionSig, CallType calltype, bytes memory initData) =
+            abi.decode(params, (bytes4, CallType, bytes));
+        if (_isFallbackHandlerInstalled(functionSig)) revert();
+
+        $moduleManager().$fallbacks[functionSig] = FallbackHandler(handler, calltype);
         IFallback(handler).onInstall(initData);
```

Comment:
if calltype is delegatecall, should `onInstall()` be called with delegatecall?

Reply:
good shout!